rclcpp/include/rclcpp/subscription_options.hpp

```diff
@@ -54,6 +55,24 @@ struct SubscriptionOptionsBase
   /// Optional RMW implementation specific payload to be used during creation of the subscription.
   std::shared_ptr<rclcpp::detail::RMWImplementationSpecificSubscriptionPayload>
   rmw_implementation_payload = nullptr;
+
+  // Options to configure topic statistics collector in the subscription.
+  struct TopicStatisticsOptions
+  {
+    // Represent the state of topic statistics collector.
+    enum class TopicStatisticsState {ENABLED, DISABLED};
```

Comment:
I wonder if it would not be wise to have a "node default" option here, like we do with intra process communication. The use case in ROS 1 is that you'd turn on statistics globally with the flip of a configuration. But it would also be nice to be able to explicitly turn statistics on or off for a subscription, overriding the default.

Reply:
https://github.com/ros2/rclcpp/blob/44fa4fe019c1c3f27c3f800fae8f465d827dfec0/rclcpp/include/rclcpp/subscription_options.hpp#L48-L49

Reply:
Interesting. I think it'd be nice to have the ability to enable topic statistics per node/subscription as well. As for whether we put it in this PR or not - given the distro freeze  date I'd prefer merging this PR as it is, if per-node configuration is not a strict requirement. I'd be happy to follow up with another PR to add a NodeDefault option.

Reply:
@wjwwood could we do this as a follow up?

Reply:
Sure.

Is there some demos or something that shows how you plan on it being used in practice? My concern is that this will need to change and break API to address any ergonomics, but we can wait and see for a bit longer. My thinking is that this change will be desired to make it convenient to use, but I could be wrong.

Reply:
We have on our roadmap to add a Topic Statistics tutorial to the ROS 2 Wiki (https://github.com/ros-tooling/aws-roadmap/issues/246) as well as adding it to the Concepts section (https://github.com/ros-tooling/aws-roadmap/issues/245). We talked about a possible demo (e.g. ROS2 demos) but we're hoping that could be done after the API freeze.

Reply:
I also have a partial implementation for adding `NodeOptions` that I will put up for review soon, just don't want to block this PR on that so we can move ahead with other topic statistics tasks in parallel.